### PR TITLE
Include Executable WithArgs in manifest

### LINKED ISSUE #1362: Args added to executable resources via the WithArgs() extension method are not written to the manifest
e.g.:

```csharp
builder.AddExecutable(name: "myapp", command: "myapp", workingDirectory: "myappdir)
    .WithArgs("do", "the", "thing");
```

This will correctly process the args passed to `WithArgs` when running the app, but it won't include those args in the manifest.

### PR DESCRIPTION
**Pull request by Kahbazi**
Fixes https://github.com/dotnet/aspire/issues/1362
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/1366)

### CODE REVIEW

File: src/Aspire.Hosting/Publishing/ManifestPublisher.cs [hunk 1]
```diff
@@ -132,13 +132,26 @@ private static void WriteExecutable(ExecutableResource executable, ManifestPubli
...
+        if (executable.TryGetAnnotationsOfType<ExecutableArgsCallbackAnnotation>(out var argsCallback))
         {
-            context.Writer.WriteStringValue(arg);
+            foreach (var callback in argsCallback)
+            {
+                callback.Callback(args);
+            }
+        }
+
+        if (args.Count > 0)
```
Comment (Kahbazi) [inline src/Aspire.Hosting/Publishing/ManifestPublisher.cs:145]: Previously if `ExecutableResource.Args` was empty the "args" would be in the manifest with an empty array. I changed this to not include the `args` to be consistence with Container manifest.

File: src/Aspire.Hosting/Extensions/ExecutableResourceBuilderExtensions.cs [hunk 2]
```diff
@@ -40,6 +40,22 @@ public static IResourceBuilder<ExecutableResource> AddExecutable(this IDistribut
...
     }
 
+    /// <summary>
+    /// Adds the arguments to be passed to an executable resource when the executable is started.
+    /// </summary>
+    /// <typeparam name="T">The resource type.</typeparam>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="args">The arguments to be passed to the executable when it is started.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<T> WithArgs<T>(this IResourceBuilder<T> builder, params string[] args) where T : ExecutableResource
```
Comment (Kahbazi) [inline src/Aspire.Hosting/Extensions/ExecutableResourceBuilderExtensions.cs:50]: I copied this from https://github.com/dotnet/aspire/pull/1363 to be able to produce the bug and write test for it.

File: src/Aspire.Hosting/Publishing/ManifestPublisher.cs [hunk 3]
```diff
@@ -132,13 +132,26 @@ private static void WriteExecutable(ExecutableResource executable, ManifestPubli
...
+        {
+            context.Writer.WriteStartArray("args");
+
+            foreach (var arg in args ?? [])
```
Comment (DamianEdwards) [inline src/Aspire.Hosting/Publishing/ManifestPublisher.cs:149]: `args` will never be null here now right, so you can avoid the `??`

File: src/Aspire.Hosting/Publishing/ManifestPublisher.cs (hunk 1 above)
Comment (DamianEdwards) [reply src/Aspire.Hosting/Publishing/ManifestPublisher.cs:145]: Add a test to verify this?

File: tests/Aspire.Hosting.Tests/ManifestGenerationTests.cs [hunk 4]
```diff
@@ -103,6 +103,32 @@ public void EnsureContainerWithArgsEmitsContainerArgs()
             arg => Assert.Equal("more", arg.GetString()));
     }
 
+    [Fact]
+    public void EnsureExecutableWithArgsEmitsExecutableArgs()
```
Comment (DamianEdwards) [inline tests/Aspire.Hosting.Tests/ManifestGenerationTests.cs:107]: Perhaps add a variant that ensures args are added via the extension method even if the resource itself had no args added.

File: src/Aspire.Hosting/Publishing/ManifestPublisher.cs (hunk 1 above)
Comment (Kahbazi) [reply src/Aspire.Hosting/Publishing/ManifestPublisher.cs:145]: Done.

File: src/Aspire.Hosting/Publishing/ManifestPublisher.cs (hunk 3 above)
Comment (Kahbazi) [reply src/Aspire.Hosting/Publishing/ManifestPublisher.cs]: Done.

File: tests/Aspire.Hosting.Tests/ManifestGenerationTests.cs (hunk 4 above)
Comment (Kahbazi) [reply tests/Aspire.Hosting.Tests/ManifestGenerationTests.cs]: Done.

**Review by mitchdenny — APPROVED** [review]
This looks good.

**Comment by mitchdenny** [comment]
Thanks for the contribution @Kahbazi!

**Pull request merged**